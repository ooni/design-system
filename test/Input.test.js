/* global describe, test, expect, jest */
import React from 'react'
import { fireEvent } from '@testing-library/react'

import { renderWithTheme } from './index'
import { Input } from '../'

describe('Input', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <Input error='Some error' />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders with prop type', () => {
    const { container } = renderWithTheme(
      <Input type='number' />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders with prop rows', () => {
    const { container } = renderWithTheme(
      <Input type='textarea' rows={4} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders with prop fontSize', () => {
    const { container } = renderWithTheme(
      <Input fontSize={4} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders with onChange handler', () => {
    const value = 'OONI'
    const onChange = jest.fn()
    const { container, getByTestId } = renderWithTheme(
      <Input onChange={onChange} data-testid='test-input'/>
    )

    expect(container.firstChild).toMatchSnapshot()

    const input = getByTestId('test-input')
    fireEvent.change(input, {
      target: { value }
    })
    expect(input.value).toBe(value)
    expect(onChange).toHaveBeenCalledTimes(1)
  })
})
