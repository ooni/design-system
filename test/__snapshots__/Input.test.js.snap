// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: block;
  vertical-align: middle;
  background-color: transparent;
  border: none;
  border-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  outline: none;
  margin: 0;
  padding: 0;
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
  border-bottom-color: #f03e3e;
  font-size: 16px;
}

.c0:hover {
  border-bottom-color: #f03e3e;
}

.c0:focus {
  border-bottom-color: #f03e3e;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 16px;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: #f03e3e;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  width: 16px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <input
    class="c0"
    font-size="1"
    rows="3"
    type="text"
  />
  <div
    class="c1 c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Some error
      </div>
    </div>
    <div
      class="c5"
      width="16"
    >
      <div
        class="c4"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Input renders with onChange handler 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: block;
  vertical-align: middle;
  background-color: transparent;
  border: none;
  border-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  outline: none;
  margin: 0;
  padding: 0;
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
  border-bottom-color: #f1f3f5;
  font-size: 16px;
}

.c0:hover {
  border-bottom-color: #f1f3f5;
}

.c0:focus {
  border-bottom-color: #0588cb;
}

<div>
  <input
    class="c0"
    data-testid="test-input"
    font-size="1"
    rows="3"
    type="text"
  />
</div>
`;

exports[`Input renders with prop fontSize 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: block;
  vertical-align: middle;
  background-color: transparent;
  border: none;
  border-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  outline: none;
  margin: 0;
  padding: 0;
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
  border-bottom-color: #f1f3f5;
  font-size: 36px;
}

.c0:hover {
  border-bottom-color: #f1f3f5;
}

.c0:focus {
  border-bottom-color: #0588cb;
}

<div>
  <input
    class="c0"
    font-size="4"
    rows="3"
    type="text"
  />
</div>
`;

exports[`Input renders with prop rows 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: block;
  vertical-align: middle;
  background-color: transparent;
  border: none;
  border-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  outline: none;
  margin: 0;
  padding: 0;
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
  border-bottom-color: #f1f3f5;
  font-size: 16px;
}

.c0:hover {
  border-bottom-color: #f1f3f5;
}

.c0:focus {
  border-bottom-color: #0588cb;
}

<div>
  <textarea
    class="c0"
    font-size="1"
    rows="4"
    type="textarea"
  />
</div>
`;

exports[`Input renders with prop type 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: block;
  vertical-align: middle;
  background-color: transparent;
  border: none;
  border-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  outline: none;
  margin: 0;
  padding: 0;
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
  border-bottom-color: #f1f3f5;
  font-size: 16px;
}

.c0:hover {
  border-bottom-color: #f1f3f5;
}

.c0:focus {
  border-bottom-color: #0588cb;
}

<div>
  <input
    class="c0"
    font-size="1"
    rows="3"
    type="number"
  />
</div>
`;
